src/Uno.UWP/ApplicationModel/Appointments/AppointmentStore.Android.cs

```diff
@@ -0,0 +1,240 @@
+﻿#nullable enable
+
+#if __ANDROID__
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Windows.Foundation;
+
+namespace Windows.ApplicationModel.Appointments
+{
+	public partial class AppointmentStore
+	{
+		// set to `true` if should be included in UWP output result set
+		private bool _startTimeRequested = false;
+		private bool _durationRequested = false;
+
+		private List<string> UWP2AndroColumnNames(IList<string> uwpColumns)
+		{
+			_startTimeRequested = false;
+
+			List<string> androidColumns = new ();
+			foreach (var column in uwpColumns)
+			{
+				switch (column)
+				{
+					case "Appointment.AllDay":
+						androidColumns.Add(Android.Provider.CalendarContract.IEventsColumns.AllDay);
+						break;
+					case "Appointment.Location":
+						androidColumns.Add(Android.Provider.CalendarContract.IEventsColumns.EventLocation);
+						break;
+					case "Appointment.StartTime":
+						_startTimeRequested = true;
+						break;
+					case "Appointment.Duration":
+						_durationRequested = true;
+						break;
+					case "Appointment.Subject":
+						androidColumns.Add(Android.Provider.CalendarContract.IEventsColumns.Title);
+						break;
+					case "Appointment.Organizer":
+						androidColumns.Add(Android.Provider.CalendarContract.IEventsColumns.Organizer);
+						break;
+					case "Appointment.Details":
+						androidColumns.Add(Android.Provider.CalendarContract.IEventsColumns.Description);
+						break;
+					case "Appointment.Reminder":
+						androidColumns.Add(Android.Provider.CalendarContract.IEventsColumns.HasAlarm);
+						break;
+
+				}
+			}
+			return androidColumns;
+		}
+
+		public IAsyncOperation<IReadOnlyList<Appointment>> FindAppointmentsAsync(DateTimeOffset rangeStart, TimeSpan rangeLength, FindAppointmentsOptions options)
+			=> FindAppointmentsAsyncTask(rangeStart, rangeLength, options).AsAsyncOperation<IReadOnlyList<Appointment>>();
+
+		// I don't know if this is default everywhere, or only on my tablet...
+		private int DEFAULT_REMINDER_MINUTES = 15;
+
+		private async Task<IReadOnlyList<Appointment>> FindAppointmentsAsyncTask(DateTimeOffset rangeStart, TimeSpan rangeLength, FindAppointmentsOptions options)
+		{
+			List<Appointment> entriesList = new List<Appointment>();
+
+			if (options is null)
+			{
+				throw new ArgumentNullException("Windows.ApplicationModel.Appointments.AppointmentStore.FindAppointmentsAsyncTask, options cannot be null");

```

Comment:
Don't use `ArgumentNullException`, use `InvalidOperationException` for all the file.

Reply:
changing to 'return null', to be in pair with UWP behaviour (although harder to catch errors :) ).